app/src/main/java/social/entourage/android/message/push/GcmBroadcastReceiver.java

```diff
@@ -1,29 +1,69 @@
 package social.entourage.android.message.push;
 
-import android.content.BroadcastReceiver;
-import android.content.Context;
 import android.content.Intent;
-import android.os.Bundle;
+import android.content.SharedPreferences;
 
-import com.google.android.gms.gcm.GoogleCloudMessaging;
+import com.google.firebase.messaging.FirebaseMessagingService;
+import com.google.firebase.messaging.RemoteMessage;
 import com.mixpanel.android.mpmetrics.GCMReceiver;
 
-public class GcmBroadcastReceiver extends BroadcastReceiver {
+import java.util.Map;
+
+import social.entourage.android.EntourageApplication;
+import social.entourage.android.api.model.Message;
+import social.entourage.android.api.tape.Events;
+import social.entourage.android.tools.BusProvider;
+import timber.log.Timber;
+
+public class GcmBroadcastReceiver extends FirebaseMessagingService {
```

Comment:
You should rename this receiver to FirebaseReceiver or FcmBroadcastReceiver to not mistake readers

Reply:
Done. New name: EntourageFirebaseMessagingService